### PR TITLE
Fix Horovod build error due to missing exported symbols

### PR DESCRIPTION
**Pull request by apeforest**
## Description ##
Export MXAPIHandle symbol and fixes https://github.com/apache/incubator-mxnet/issues/17292

Here are the causes of the problem:

1.  Horovod uses MX_API_BEGIN() and MX_API_END() from mxnet/c_api_error.h to catch and throw errors in horovod APIs: https://github.com/horovod/horovod/blob/master/horovod/mxnet/mpi_ops.cc#L224
MX_API_BEGIN() is a macro that calls MXAPIHandleException https://github.com/apache/incubator-mxnet/blob/master/include/mxnet/c_api_error.h#L36. Before #17128, MXAPIHandleException is an inline function. And therefore when #17128 introduced a new function call NormalizeError() inside MXAPIHandleException it broke Horovod integration because the symbol of NormalizeError is not whitelist by MXNet distribution.
2. #17298 removed NormalizeError() from MXAPIHandleException and made it not inline. https://github.com/apache/incubator-mxnet/pull/17208/files#diff-875aa4c013dbd73b044531e439e8afddR67. This time the error becomes undefined symbol of MXAPIHandleException.

So to summarize, the problem is not that Horovod requires MXAPIHandleException function to be inline. The rootcause is that MXNet did not whitelist the symbol *MXAPIHandleException* but only whitelisted the symbols that are being used inside MXAPIHandleException function. It was okay when the function MXAPIHandleException is inline, but became a problem when it's not. A good practice is to whitelist symbol *MXAPIHandleException* instead of its internal functions.
## Comments ##
@szha Please help to review.

### CODE REVIEW

File: make/config/libmxnet.sym [hunk 1]
```diff
@@ -12,4 +12,4 @@ Java_org_apache_mxnet*
 *Storage*Get*
 *on_enter_api*
 *on_exit_api*
-*MXAPISetLastError*
+*MXAPIHandleException*
```
Comment (szha) [inline make/config/libmxnet.sym]: Isn't this covered by the first line? If not, should we expand the pattern for the first line instead?

Comment (apeforest) [reply make/config/libmxnet.sym]: I believe the first line misses the prefix before MX. Not sure if we want to whitelist all *MX* APIs.

Comment (szha) [reply make/config/libmxnet.sym]: Let's fix the first line then

Comment (apeforest) [reply make/config/libmxnet.sym]: Updated. Please review again. Thanks!

**Comment by apeforest** [comment]
@szha Thanks for the review and update. Please let me know if this is good to ship.
